airbyte-workers/src/main/resources/dbt_transformation_entrypoint.sh

```diff
@@ -2,48 +2,54 @@
 set -e
 
 CWD=$(pwd)
-if [[ -f "${CWD}/git_repo/dbt_project.yml" ]]; then
-  # Find profile name used in the custom dbt project:
-  PROFILE_NAME=$(grep -e "profile:" < "${CWD}/git_repo/dbt_project.yml" | sed -E "s/profile: *['\"]?([^'\"]*)['\"]?/\1/")
-  if [[ -n "${PROFILE_NAME}" ]]; then
-    mv "${CWD}/profiles.yml" "${CWD}/profiles.txt"
-    # Refer to the appropriate profile name in the profiles.yml file
-    echo "Regenerate profiles.yml file for profile: $PROFILE_NAME"
-    sed -E "s/normalize:/$PROFILE_NAME:/" < "${CWD}/profiles.txt" > "${CWD}/profiles.yml"
-    rm -f "${CWD}/profiles.txt"
-    if [[ -f "${CWD}/bq_keyfile.json" ]]; then
-      cp "${CWD}/bq_keyfile.json" /tmp/bq_keyfile.json
-    fi
-  fi
-else
-  echo "git repo does not contain a dbt_project.yml file?"
-fi
+# change directory to be inside git_repo that was just cloned
+cd git_repo
+
+POSITIONAL=()
 # Detect if some mandatory dbt flags were already passed as arguments
 CONTAINS_PROFILES_DIR="false"
 CONTAINS_PROJECT_DIR="false"
-for var in "$@"; do
-    if [[ -n $(echo "${var}" | grep -e "--profiles-dir=") ]]; then
+while [[ $# -gt 0 ]]; do
+  case $1 in
+    --profiles-dir=*|--profiles-dir)
       CONTAINS_PROFILES_DIR="true"
-    fi
-    if [[ -n $(echo "${var}" | grep -e "--project-dir=") ]]; then
+      POSITIONAL+=("$1")
+      shift
+      ;;
+    --project-dir=*|--project-dir)
       CONTAINS_PROJECT_DIR="true"
-    fi
+      POSITIONAL+=("$1")
+      shift
+      ;;
+    *)
+      POSITIONAL+=("$1")
+      shift
+      ;;
+  esac
 done
+
+set -- "${POSITIONAL[@]}"
+
+if [[ -f "${CWD}/bq_keyfile.json" ]]; then
+  cp "${CWD}/bq_keyfile.json" /tmp/bq_keyfile.json
+fi
+
 # Add mandatory flags profiles-dir and project-dir when calling dbt when necessary
-if [[ "${CONTAINS_PROFILES_DIR}" = "true" ]]; then
-  if [[ "${CONTAINS_PROJECT_DIR}" =  "true" ]]; then
+case "${CONTAINS_PROFILES_DIR}-${CONTAINS_PROJECT_DIR}" in
+  true-true)
     echo "Running: dbt $@"
-    dbt $@
-  else
+    dbt $@ --profile normalize
```

Comment:
Because we use the `--profile` flag: https://github.com/dbt-labs/dbt/blob/35645a7233f4d32167d4db4a6066297cd8f79514/core/dbt/main.py#L290

We don't need to manipulate our profiles.yml to match with the declared profile in the `dbt_project.yml` file anymore... So it closes https://github.com/airbytehq/airbyte/issues/4734